tests/clickhouse-test

```diff
@@ -191,59 +204,89 @@ def need_retry(args, stdout, stderr, total_time):
         FROM clusterAllReplicas('test_cluster_database_replicated', system.processes)
         WHERE query NOT LIKE '%system.processes%'
-        """)
+        """,
+        )
     else:
-        return clickhouse_execute_json(args, 'SHOW PROCESSLIST')
+        return clickhouse_execute_json(args, "SHOW PROCESSLIST")
+
 
 def get_transactions_list(args):
     try:
         if args.replicated_database:
-            return clickhouse_execute_json(args, """
+            return clickhouse_execute_json(
+                args,
+                """
             SELECT materialize((hostName(), tcpPort())) as host, *
             FROM clusterAllReplicas('test_cluster_database_replicated', system.transactions)
-            """)
+            """,
+            )
         else:
-            return clickhouse_execute_json(args, 'select * from system.transactions')
+            return clickhouse_execute_json(args, "select * from system.transactions")
     except Exception as e:
         return f"Cannot get list of transactions: {e}"
 
+
 def get_processlist_after_test(args):
     log_comment = args.testcase_basename
     database = args.testcase_database
-    if args.replicated_database:
-        return clickhouse_execute_json(args, f"""
-        SELECT materialize((hostName(), tcpPort())) as host, *
-        FROM clusterAllReplicas('test_cluster_database_replicated', system.processes)
-        WHERE
-            query NOT LIKE '%system.processes%' AND
-            Settings['log_comment'] = '{log_comment}' AND
-            current_database = '{database}'
-        """)
-    else:
-        return clickhouse_execute_json(args, f"""
-        SELECT *
-        FROM system.processes
-        WHERE
-            query NOT LIKE '%system.processes%' AND
-            Settings['log_comment'] = '{log_comment}' AND
-            current_database = '{database}'
-        """)
+    for _ in range(3):
+        if args.replicated_database:
+            data = clickhouse_execute_json(
+                args,
+                f"""
+            SELECT materialize((hostName(), tcpPort())) as host, *
+            FROM clusterAllReplicas('test_cluster_database_replicated', system.processes)
+            WHERE
+                query NOT LIKE '%system.processes%' AND
+                Settings['log_comment'] = '{log_comment}' AND
+                current_database = '{database}'
+            """,
+            )
+        else:
+            data = clickhouse_execute_json(
+                args,
+                f"""
+            SELECT *
+            FROM system.processes
+            WHERE
+                query NOT LIKE '%system.processes%' AND
+                Settings['log_comment'] = '{log_comment}' AND
+                current_database = '{database}'
+            """,
+            )
+        if data:
+            for query_desc in data:
+                # very short queries after test may left for some small period of
+                # time
```

Comment:
I want to avoid adding such logic here.

Once client exited no queries should be left, this was previously possible in case of errors, but should not happens anymore after recent fixes, plus #36587

What do you think?